src/__tests__/types.test.ts

```diff
@@ -108,19 +108,22 @@ describe("definedValues", () => {
       c: {
         d: [1, 2],
         e: undefined
-      }
+      },
+      f: [{ g: 1, h: undefined }, { i: undefined }]
     };
 
     const newObj = withoutUndefinedValues(obj);
 
-    expect(Object.keys(newObj).length).toEqual(2);
+    expect(Object.keys(newObj).length).toEqual(3);
     expect(Object.keys(newObj.c).length).toEqual(1);
+    expect(newObj.f.length).toEqual(1);
```

Comment:
```suggestion
```

since you're comparing the whole payload I guess we can omit this check.

Reply:
yes, and I think we can remove all the checks on the length of values and check the output using the [`toStrictEqual`](https://jestjs.io/docs/expect#tostrictequalvalue) method